### DISCUSSION TITLE
Timetables: logical_date (deprecated execution_date) does not align with data_interval.start for manual DAG trigger

### DISCUSSION THREAD

**Discussion by sushinoya** [apache/airflow#21577] [Q&A]
### Apache Airflow version

2.2.3 (latest released)

### What happened

I created a simple DAG using the `CronDataIntervalTimetable`. I used the Trigger DAG button on the UI to trigger a DAG Run. The cron expression that I was using was `5,10,15,20,25  * * * *`  and I pressed the button at `13:18`.

I expected the Data Interval to be `13.10` to `13.15` and that was represented correctly on the UI -

<img width="300" alt="Screenshot 2022-02-15 at 1 21 29 PM" src="https://user-images.githubusercontent.com/23443586/153997693-f5d44c35-51ac-4124-8a96-7125f57879c1.png">

I also expected the `execution_date` a.k.a `logical_date` to be equal to `data_interval.start` i.e `13.10`. However it was `13.18` instead (or `05:18` in UTC as shown in the image).

<img width="600" alt="Screenshot 2022-02-15 at 1 23 58 PM" src="https://user-images.githubusercontent.com/23443586/153998009-acffcbca-127a-4f31-ab06-2e3f8688574e.png">

The same can be seen from the Task Instance Details tab of the `BashOperator`.

<img width="500" alt="Screenshot 2022-02-15 at 1 25 26 PM" src="https://user-images.githubusercontent.com/23443586/153998072-c6518232-109a-4d10-b873-8423a89e2b9e.png">

### What you expected to happen

As explained above, I expected the `execution_date` to be equal to the `data_interval.start`. In fact, for timetables this is how `logical_date` (i.e `execution_date`) it is defined - https://github.com/apache/airflow/blob/0cd3b11f3a5c406fbbd4433d8e44d326086db634/airflow/timetables/base.py#L93-L100

Thus it seems rather odd or it to behave differently in this case. We do know the data_interval from the Timetable's `infer_manual_data_interval` function and it is reflected correctly on the UI too. However, the `execution_date` is not updated accordingly.

### How to reproduce

Add a file `cron_data_interval_timetable_test.py` with the following contents to the dags folder. Update the timezone to your local timezone for convenience - 

```python
import datetime

from airflow import DAG
from airflow.operators.bash import BashOperator
from airflow.timetables.interval import CronDataIntervalTimetable
from pendulum.tz.timezone import Timezone

with DAG(
    dag_id="cron_data_interval_timetable_test",
    start_date=datetime.datetime(2021, 1, 1),
    timetable=CronDataIntervalTimetable("5,10,15,20,25 * * * *", Timezone("Asia/Singapore")),
    tags=["example", "timetable"],
    catchup=False
) as dag:
    BashOperator(
    task_id="print_day_of_week",
    bash_command="echo Execution Date is {{ execution_date }}",
)
```

Enable this DAG on the UI and trigger a manual run using the play button on the top right. Then look into the DAG run's `print_day_of_week` tasks's Rendered Template and Task Instance Details. Both mention execution date.

### Operating System

MacOS Big Sur (11.6.1)

### Deployment

Docker-Compose

### Deployment details

- `curl -LfO 'https://airflow.apache.org/docs/apache-airflow/2.2.3/docker-compose.yaml'`
- `docker compose up`

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by uranusjr** [comment]
This is expected. Unfortunately execution_date pointed to the _run date_ before timetable was introduced as a concept, and we had to keep backward compatibility. The only viable alternative would be to keep execution_date point to the run date, but have logical date match data_interval.start, but that is arguably even more confusing. Ultimately logical date (and execution_date) does not really mean anything semantically, but only identifies the given DAG run. It was so, is so, and will be so (until Airflow 3.0 when we can break compatibility, but at that point I'd hope logical date cease to exist entirely). I am sorry, but it is what it is.

**Reply by sushinoya** [reply]
Hi @uranusjr, thank you for the quick response! Although not ideal, I actually prefer your alternative of at least having logical date reflecting the correct value. I say this because we are already on our way to deprecate execution_date. And if we have no choice but to have an inconsistency to be backwards compatible, we can have the newly introduced concepts (in this case logical date) be consistent.

**Reply by potiuk** [reply]
We are adding some more decriptions regarding the "logical_date" concept. As @uranusjr - it has only meaning in case of the same use cases that were there in the first place - so in case you have data intervals which were "ending". 

I fully agre with TP that  it would be even more confusing if we change it to interval_start. 
And there is really no point in doing it. If you want to use "interval_start" - welll... use "interval_start" and forget about logical_date. If you really care about "start" "end" - just use those. 

The "run_id" is far better to distinguish between different runs, "interval_start", "interval_end" are there too. Simply don't use "logical_date" why would you care what it is ?

**Reply by potiuk** [reply]
BTW. Using "data_interval_start" instead of "logical_date" or "execution_date' in case you care about interval - is just perfect embodiment of "explicit is better than implicit" zen of Python. It is, what it is. and there is no argument nor confusion what it is. 

It just requires you to mentally switch to use it in your conversations, design etc. .

**Reply by nicchristeller** [reply]
I just want to say that, as an AirFlow beginner, this was exceedingly confusing.

[The docs](https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#data-interval) state that "The logical date (also called execution_date in Airflow versions prior to 2.2) of a DAG run, for example, denotes the start of the data interval, not when the DAG is actually executed."

But this clearly isn't always the case.

**Reply by potiuk** [reply]
Do you have any suggestions how to fix it? Airlfow is created by > 2100 contributors and more often than not people provide updates to docuemntation not only the code that makes it clearer for others who come to understand Airflow concepts and docs. Actuallly contributing doc updates to improve the description is one of the best ways users who just struggled with it can contribute back - because long time maintainrs have too many assumptions in their head. 

Airflow is free software. Anyone using it paid exactly 0 to use it. I think you are  probably one of the best people in the world to update the documentation in the way that it will be clearer for people like you because you know what you struggled with and you know where you looked for documentation and what assumptions you missed.

And it's super easy - just click on the "suggest a chnage on this page" when you go to [The docs](https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#data-interval). This will open a PR in GitHub UI that is as easy as wrting the comment above - only instead of writing a complaining comment, you can contribute update that will be useful by others.

**Reply by uranusjr** [reply]
I would say just get rid of that paragraph in the documentation. If we want to replace it with something, it should be along the lines of

1. Use `data_interval` variables instead.
2. `logical_date` (and `execution_date`) doesn’t inheritly mean anything, but only serve as an arbitrary value that “represents” the DAG run. It can be used to distinguish a specific DAG run from another (although note that they are _not necessarily unique_), nothing more.
